subprojects/logging/src/integTest/groovy/org/gradle/DeprecationHandlingIntegrationTest.groovy

```diff
@@ -129,6 +135,31 @@ class DeprecationHandlingIntegrationTest extends AbstractIntegrationSpec {
         'with stacktrace and --warning-mode=no'         | WarningMode.None    | 0                      | 0                      | true
         'without stacktrace and --warning-mode=summary' | WarningMode.Summary | 0                      | 4                      | false
         'with stacktrace and --warning-mode=summary'    | WarningMode.Summary | 0                      | 4                      | true
+        'without stacktrace and --warning-mode=fail'    | WarningMode.Fail    | 4                      | 0                      | false
+        'with stacktrace and --warning-mode=fail'       | WarningMode.Fail    | 4                      | 0                      | true
+    }
+
+    def 'build error and deprecation failure combined'() {
+        given:
+        buildFile << """
+            apply plugin: DeprecatedPlugin // line 2
+            
+            task broken() {
+                doLast {
+                    throw new IllegalArgumentException("Can't do that")
+                }
+            }
+        """.stripIndent()
```

Comment:
minor: is this displayed anywhere, or what's the reason for the strip?

Reply:
I applied patterns from elsewhere in that test class. I believe it will remove the leading spaces thus resulting in a normal looking build file, though we rarely look at them.